furusiya/view/renderer.py

```diff
@@ -0,0 +1,149 @@
+        else:
+            Game.target = None
+            x2, y2 = Game.mouse_coord
+        x1, y1 = Game.player.x, Game.player.y
+        line = tdl.map.bresenham(x1, y1, x2, y2)
+        for pos in line:
+            Game.con.draw_char(pos[0], pos[1], '*', colors.dark_green, bg=None)
+        tdl.flush()
+
+        # Undo drawing tiles outside sight
+        for pos in line:
+            if not pos in Game.visible_tiles:
+                Game.con.draw_char(pos[0], pos[1], '#', colors.black)
+
+    # draw all objects in the list
+    for obj in Game.area_map.entities:
+        if obj != Game.player:
+            if Game.draw_bowsight and (obj.x, obj.y) in Game.visible_tiles and \
+                            (obj.x, obj.y) == (x2, y2) and obj.get_component(Fighter) is not None:
+                Game.target = obj
+                Game.con.draw_char(obj.x, obj.y, 'X', fg=colors.red)
+            else:
+                obj.draw()
+
+    Game.player.draw()
+    Game.stallion.draw()
+
+    # blit the contents of "Game.con" to the root console and present it
+    Game.root.blit(Game.con, 0, 0, MAP_WIDTH, MAP_HEIGHT, 0, 0)
+
+    # prepare to render the GUI Game.panel
+    Game.panel.clear(fg=colors.white, bg=colors.black)
+
+    # print the game messages, one line at a time
+    y = 1
+    for (line, color) in Game.game_msgs:
+        Game.panel.draw_str(MSG_X, y, line, bg=None, fg=color)
+        y += 1
+
+    # show the player's stats
+    player_fighter = Game.player.get_component(Fighter)
+    Game.panel.draw_str(1, 1, "HP: {}/{}".format(player_fighter.hp, player_fighter.max_hp))
+    # render_bar(1, 1, BAR_WIDTH, 'HP', player.fighter.hp, player.fighter.max_hp,
+    #     colors.light_red, colors.darker_red)
+
+    # display names of objects under the mouse
+    Game.panel.draw_str(1, 0, get_names_under_mouse(), bg=None, fg=colors.light_gray)
+
+    # blit the contents of "Game.panel" to the root console
+    Game.root.blit(Game.panel, 0, PANEL_Y, SCREEN_WIDTH, PANEL_HEIGHT, 0, 0)
+
+    tdl.flush()
+
+
+def is_visible_tile(x, y):
+
+    if x >= MAP_WIDTH or x < 0:
+        return False
+    elif y >= MAP_HEIGHT or y < 0:
+        return False
+    elif not Game.area_map.tiles[x][y].is_walkable:
+        return False
+    elif Game.area_map.tiles[x][y].block_sight:
+        return False
+    else:
+        return True
+
+
+def get_names_under_mouse():
+    # create a list with the names of all objects at the mouse's coordinates and in FOV
+    names = [obj.name for obj in get_objects_under_mouse()]
+
+    names = ', '.join(names)  # join the names, separated by commas
+    return names.capitalize()
+
+
+def get_objects_under_mouse():
+    (x, y) = Game.mouse_coord
+
+    # create a list with the names of all objects at the mouse's coordinates and in FOV
+    stuff = [obj for obj in Game.area_map.entities
+             if obj.x == x and obj.y == y and (obj.x, obj.y) in Game.visible_tiles]
+
+    return stuff
+
+
+def closest_monster(max_range):
```

Comment:
Ditto here (targetting functions deal with mouse, or targetting by distance -- maybe we can also split it along those lines into two files)

Reply:
Moving them would be easier if the code wasn't spaghetti.
It's hard not to fall into cyclic imports. They will be moved when the code is clean enough.